### PR TITLE
ghidra,ghidra-bin: update to 10.3

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Description of changes

Motivation: https://github.com/NixOS/nixpkgs/issues/193370

Motivation 2: there is now a dark theme 😎 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erdnaxe** [comment]
Result of `nixpkgs-review pr 232882` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ghidra</li>
    <li>ghidra-bin</li>
  </ul>
</details>

**Comment by tnias** [comment]
I tested the new theming and basic things for ghidra and ghidra-bin on x86_64-linux. Works for me. :+1: 
Thanks. :)

**Comment by RaitoBezarius** [comment]
Please fix the conflicts

**Comment by erdnaxe** [comment]
> Please fix the conflicts

Thanks for the ping, conflicts should now be fixed.

Darwin is failing, but that was already the case before this pull request.

File: pkgs/tools/security/ghidra/default.nix [hunk 1]
```diff
@@ -24,12 +24,12 @@ let
...
   pname = "ghidra";
-  version = "10.2.3";
-  versiondate = "20230208";
+  version = "10.3";
+  versiondate = "20230510";
 
   src = fetchzip {
     url = "https://github.com/NationalSecurityAgency/ghidra/releases/download/Ghidra_${version}_build/ghidra_${version}_PUBLIC_${versiondate}.zip";
-    hash = "sha256-0uDLS+fnUqLw+oqoR/Vs56vhzNmA54k0WBvhvDmlGKs=";
+    sha256 = "sha256-uFyTMWhj3yMVIPxEwkLtTqpJUi2S8A2GFjjY3rNTC2c=";
```
Comment (RaitoBezarius) [inline pkgs/tools/security/ghidra/default.nix:32]: why the rename in sha256?

File: pkgs/tools/security/ghidra/build.nix [hunk 2]
```diff
@@ -12,20 +12,20 @@
...
   pname = "ghidra";
-  version = "10.2.3";
+  version = "10.3";
 
   src = fetchFromGitHub {
     owner = "NationalSecurityAgency";
     repo = "Ghidra";
     rev = "Ghidra_${version}_build";
-    sha256 = "sha256-YhjKRlFlF89H05NsTS69SB108rNiiWijvZZY9fR+Ebc=";
+    sha256 = "sha256-v3XP+4fwjPzt/OOxX27L0twXw8T1Y94hgP4A5Ukol5I=";
```
Comment (RaitoBezarius) [inline pkgs/tools/security/ghidra/build.nix:28]: please use hash

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 232882` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ghidra</li>
    <li>ghidra-bin</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 232882` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ghidra</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 232882` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ghidra</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ghidra-bin</li>
  </ul>
</details>

File: pkgs/tools/security/ghidra/default.nix (hunk 1 above)
Comment (erdnaxe) [reply pkgs/tools/security/ghidra/default.nix]: Oops, my bad. Fixed!

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 232882` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ghidra</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
```
ghidra-aarch64-darwin> Successfully started process 'Gradle build daemon'
ghidra-aarch64-darwin> An attempt to start the daemon took 0.503 secs.
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> FAILURE: Build failed with an exception.
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> * What went wrong:
ghidra-aarch64-darwin> Unable to start the daemon process.
ghidra-aarch64-darwin> This problem might be caused by incorrect configuration of the daemon.
ghidra-aarch64-darwin> For example, an unrecognized jvm option is used.
ghidra-aarch64-darwin> Please refer to the User Manual chapter on the daemon at https://docs.gradle.org/7.6.1/userguide/gradle_daemon.html
ghidra-aarch64-darwin> Process command line: /nix/store/vznwx8md3ma3nfviy23hl7905n1hmn6k-zulu17.34.19-ca-jdk-17.0.3/bin/java --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED --add-opens=java.base/java.nio.charset=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED -Xmx2G -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp /nix/store/n4mjqb2pa0bkvahaww540jln9rdkrk5f-gradle-7.6.1/lib/gradle/lib/gradle-launcher-7.6.1.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 7.6.1
ghidra-aarch64-darwin> Please read the following process output to find out more:
ghidra-aarch64-darwin> -----------------------
ghidra-aarch64-darwin> Picked up JAVA_TOOL_OPTIONS: -Duser.home='/private/tmp/nix-build-ghidra-10.3.drv-0/home'
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> FAILURE: Build failed with an exception.
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> * What went wrong:
ghidra-aarch64-darwin> java.net.SocketException: Operation not permitted
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> * Try:
ghidra-aarch64-darwin> > Run with --info or --debug option to get more log output.
ghidra-aarch64-darwin> > Run with --scan to get full insights.
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> * Exception is:
ghidra-aarch64-darwin> org.gradle.api.UncheckedIOException: java.net.SocketException: Operation not permitted
ghidra-aarch64-darwin>         at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:62)
ghidra-aarch64-darwin>         at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:41)
ghidra-aarch64-darwin>         at org.gradle.internal.remote.internal.inet.TcpIncomingConnector.accept(TcpIncomingConnector.java:62)
ghidra-aarch64-darwin>         at org.gradle.launcher.daemon.server.DaemonTcpServerConnector.start(DaemonTcpServerConnector.java:86)
ghidra-aarch64-darwin>         at org.gradle.launcher.daemon.server.Daemon.start(Daemon.java:169)
ghidra-aarch64-darwin>         at org.gradle.launcher.daemon.bootstrap.DaemonMain.doAction(DaemonMain.java:125)
ghidra-aarch64-darwin>         at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:50)
ghidra-aarch64-darwin>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
ghidra-aarch64-darwin>         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
ghidra-aarch64-darwin>         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
ghidra-aarch64-darwin>         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
ghidra-aarch64-darwin>         at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:60)
ghidra-aarch64-darwin>         at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:37)
ghidra-aarch64-darwin>         at org.gradle.launcher.daemon.bootstrap.GradleDaemon.main(GradleDaemon.java:22)
ghidra-aarch64-darwin> Caused by: java.net.SocketException: Operation not permitted
ghidra-aarch64-darwin>         at java.base/sun.nio.ch.Net.bind0(Native Method)
ghidra-aarch64-darwin>         at java.base/sun.nio.ch.Net.bind(Net.java:555)
ghidra-aarch64-darwin>         at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337)
ghidra-aarch64-darwin>         at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294)
ghidra-aarch64-darwin>         at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89)
ghidra-aarch64-darwin>         at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:81)
ghidra-aarch64-darwin>         at org.gradle.internal.remote.internal.inet.TcpIncomingConnector.accept(TcpIncomingConnector.java:59)
ghidra-aarch64-darwin>         ... 11 more
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> * Get more help at https://help.gradle.org
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> * Try:
ghidra-aarch64-darwin> > Run with --stacktrace option to get the stack trace.
ghidra-aarch64-darwin> > Run with --debug option to get more log output.
ghidra-aarch64-darwin> > Run with --scan to get full insights.
ghidra-aarch64-darwin> 
ghidra-aarch64-darwin> * Get more help at https://help.gradle.org
```

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 232882` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ghidra</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ghidra-bin</li>
  </ul>
</details>

**Comment by erdnaxe** [comment]
Marked broken on Darwin.

From what I understand, the darwin build does not require the same Gradle dependencies. Maybe someone could contribute a patch for `ghidra.deps` with a correct `outputHash`.

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**